tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_iv_index.c

```diff
@@ -77,13 +77,16 @@ static void test_ivu_recovery(void)
 
 	atomic_clear_bit(bt_mesh.flags, BT_MESH_IVU_IN_PROGRESS);
 
-	/* Already in IV normal mode */
-	ASSERT_FALSE(bt_mesh_net_iv_update(TEST_IV_IDX + 1, BCN_IV_IN_IDLE));
+	/* Start recovery */
```

Comment:
No, this test is wrong, so it should be modified.

Reply:
Yep, but it should be modified in the correct manner. Now you skipped checking the code branch that is checked before and instead of this, it starts checking recovery that is checked below one more time. 
Your changes reduced code coverage.